Cesium.Runtime/StdIoFunctions.cs

```diff
@@ -279,6 +277,24 @@ public static int FPrintF(void* stream, byte* str, void* varargs)
 #endif
     }
 
+    internal static byte* MarshalStr(string? str)
+    {
+        Encoding encoding = Encoding.UTF8;
+        if (str is null)
+        {
+            return null;
+        }
+
+        var bytes = encoding.GetBytes(str);
+        var storage = (byte*)StdLibFunctions.Malloc((nuint)bytes.Length);
```

Comment:
Gotcha!
```suggestion
        var storage = (byte*)StdLibFunctions.Malloc((nuint)bytes.Length + 1);
```

You forgot to allocate space for the terminating null character.